codes/classical/q-ary_digits/ternary_golay.yml

```diff
@@ -0,0 +1,47 @@
+
+code_id: ternary_golay
+physical: bits
+logical: bits
+name: 'Ternary Golay Code \cite{doi:10.1007/bf03025254}'
+introduced: '\cite{manual:{Golay, Marcel JE. "Notes on digital coding." Proc. IEEE 37 (1949): 657.}}'
+description: |
+
+  The ternary Golay code \(\mathcal{G}_{11}\) is a ternary \([11,6,5]_3\) linear code with the following generator matrix: 
+  \begin{align}
+  \left[\begin{array}{ccccccccccc}
+    1 & 0 & 0 & 0 & 0 & | & 1 & 1 & 1 & 2 & 2 & 0 \\ 
+    0 & 1 & 0 & 0 & 0 & | & 1 & 1 & 2 & 1 & 0 & 2 \\ 
+    0 & 0 & 1 & 0 & 0 & | & 1 & 2 & 1 & 0 & 1 & 2 \\ 
+    0 & 0 & 0 & 1 & 0 & | & 1 & 2 & 0 & 1 & 2 & 1 \\ 
+    0 & 0 & 0 & 0 & 1 & | & 1 & 0 & 2 & 2 & 1 & 1 \\ 
+  \end{array}\right]
+  \end{align}
+
+  The ternary golay code is a perfect code because with $\(d = 5\)$, \(\mathcal{G}_11\) can correct at most up to 2 errors so the Hamming bound 
+  is \( 3^11/\sum_{k=0}^{2}{n\choose k}2^k = 729\) which is exactly the number of codewords in \(\mathcal{G}_{11}\). In fact it can be shown that any perfect 
+  code is either a Hamming code or a ternary or binary Golay code. A Golay code is also a quadratic residue code with residue set \(Q = \{1, 3, 4, 5, 9\} \) with 
```

Comment:
Two inaccuracies: first, any **linear** perfect code; second, there are also trivial perfect codes such as {00000, 11111} for any odd length n; or {0,1}^n with distance 1, for any n.